ephios/event_management/templates/event_management/home.html

```diff
@@ -1,12 +1,34 @@
 {% extends "base.html" %}
+{% load event_extras %}
 {% load i18n %}
 
 {% block title %}
     {% trans "Home" %}
 {% endblock %}
 
 {% block content %}
-<h1 class="page-header">
-    {% trans "Welcome to ephios!" %}
-</h1>
-{% endblock %}
\ No newline at end of file
+    <h1 class="page-header">
+        {% trans "Welcome to ephios!" %}
+    </h1>
+
+    {% with request.user|confirmed_shifts as confirmed_shifts %}
+        {% if confirmed_shifts.count > 0 %}
+            <div class="card mb-3">
+                <div class="card-body">
+                    <h5 class="card-title">{% trans "Your upcoming events" %}</h5>
+                </div>
+                <ul class="list-group list-group-flush">
+                    {% for shift in request.user|confirmed_shifts %}
```

Comment:
Correct me if I'm wrong, but you are querying confirmed_shifts multiple times here. Try to reduce it to one query.